app/Console/AbstractUpdateOfferStatusCommand.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Silex\Console;
+
+use Broadway\CommandHandling\CommandBusInterface;
+use CultuurNet\UDB3\Event\ValueObjects\Status;
+use CultuurNet\UDB3\Event\ValueObjects\StatusReason;
+use CultuurNet\UDB3\Event\ValueObjects\StatusType;
+use CultuurNet\UDB3\Language;
+use CultuurNet\UDB3\Offer\Commands\Status\UpdateStatus;
+use CultuurNet\UDB3\Offer\OfferType;
+use CultuurNet\UDB3\Search\ResultsGenerator;
+use CultuurNet\UDB3\Search\SearchServiceInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Question\Question;
+
+abstract class AbstractUpdateOfferStatusCommand extends AbstractCommand
+{
+    /**
+     * @var OfferType
+     */
+    protected $offerType;
+
+    /**
+     * @var ResultsGenerator
+     */
+    private $searchResultsGenerator;
+
+    public function __construct(
+        CommandBusInterface $commandBus,
+        SearchServiceInterface $searchService
+    ) {
+        parent::__construct($commandBus);
+        $this->commandBus = $commandBus;
+        $this->searchResultsGenerator = new ResultsGenerator(
+            $searchService,
+            ['created' => 'asc'],
+            100
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $query = $this->askForQuery($input, $output);
+        $count = $this->searchResultsGenerator->count($query);
+        $output->writeln("This command will update $count {$this->pluralizedOfferType()}");
+
+        $statusType = $this->askForStatusType($input, $output);
+        $reasons = $this->askForReasons($input, $output);
+
+        $status = new Status($statusType, $reasons);
+
+        $confirmation = $this->getHelper('question')
+            ->ask(
+                $input,
+                $output,
+                new ConfirmationQuestion(
+                    "This action will update the status of {$count} {$this->pluralizedOfferType()} to {$statusType->toNative()}, continue? [y/N] ",
+                    true
+                )
+            );
+
+        if (!$confirmation) {
+            return 0;
+        }
+
+        $offers = $this->searchResultsGenerator->search($query);
+        foreach ($offers as $id => $offer) {
```

Comment:
Should we show a progress bar here?

Reply:
A simple solution would be to render a dot for each finished offer?

Reply:
Or the [Symfony Progress Bar](https://symfony.com/doc/current//components/console/helpers/progressbar.html)?

Reply:
Good suggestion @LucWollants !